### PR TITLE
Extend error details in `ErrorDialog`

### PR DESCRIPTION
**Pull request by johnnyomair**
Previously, uncaught errors in production environments would result in an "An error occurred" `ErrorDialog`, making the error difficult to debug. To improve the reproducibility of an error, we enrich the `ErrorDialog` with the following `additionalInformation`:

-   `errorType`: The type of the error, network or server error
-   `httpStatus`: The HTTP status of the request that failed
-   `url`: The URL where the error occurred
-   `timestamp`: The timestamp when the error occurred

This information will be displayed in the `ErrorDialog` if no custom `userMessage` has been provided. In addition, a button has been added to allow this information to be copied to the clipboard.

---

### Examples

1. GraphQL (= server) error

https://github.com/vivid-planet/comet/assets/48853629/ea567713-c037-4fb0-9076-b497a3a9a981

2. Network error

network error

3. Session expired error

session expired

- [x] Add changeset (if necessary)

### CODE REVIEW

File: packages/admin/admin/src/error/errordialog/ErrorDialog.tsx [hunk 1]
```diff
@@ -84,3 +70,123 @@ export const ErrorDialog: React.FunctionComponent<ErrorDialogProps> = ({ show =
...
+
+const ErrorList = styled(List)`
+    list-style-type: disc;
+    padding-inline-start: ${({ theme }) => theme.spacing(6)};
+`;
+
+const ErrorListItem = styled(ListItem)`
+    display: list-item;
+    padding-left: 0;
+`;
```
Comment (johnnyomair) [inline packages/admin/admin/src/error/errordialog/ErrorDialog.tsx:150]: @thomasdax98 I've copied these from #1366. Maybe we need some general components for "normal" lists?

File: packages/admin/admin/src/error/errordialog/ErrorDialog.tsx [hunk 2]
```diff
@@ -84,3 +70,123 @@ export const ErrorDialog: React.FunctionComponent<ErrorDialogProps> = ({ show =
...
+
+type CopyToClipboardButtonProps = {
+    copyData: string;
+};
+
+function CopyToClipboardButton({ copyData }: CopyToClipboardButtonProps): JSX.Element {
```
Comment (johnnyomair) [inline packages/admin/admin/src/error/errordialog/ErrorDialog.tsx:156]: @thomasdax98 we already have a CopyToClipboardButton (introduced by @jamesricky), but it is not flexible enough for this use case (always an IconButton, no text). Maybe we could discuss the button with UX and then create a generic component for all use cases?

File: packages/admin/admin/src/error/errordialog/createErrorDialogApolloLink.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        if (networkError) {
+            // Prioritize GraphQL error over network error
+            if (!error) {
+                error = networkError.message;
+                errorType = "network";
+                title = <FormattedMessage id="comet.errorDialog.networkError.title" defaultMessage="Network error" />;
+            }
+
+            if (networkError.name === "ServerError") {
+                const statusCode = (networkError as ServerError).statusCode;
```
Comment (manuelblum) [inline packages/admin/admin/src/error/errordialog/createErrorDialogApolloLink.tsx:41]: I would avoid `as` casts and introduce a typeguard for this kind of check:

```
function isServerError(error: NetworkError | undefined): error is ServerError {
    if (error?.name === "ServerError") {
        return (error as ServerError).statusCode !== undefined;
    }
    return false;
}
```

and then using it here:

```
if (isServerError(networkError)) {
    const statusCode = networkError.statusCode;
    //....
```

Comment (johnnyomair) [reply packages/admin/admin/src/error/errordialog/createErrorDialogApolloLink.tsx]: Done: ee327ad9b6a8020a6c98b19520741f5449a4ba7d

**Review by manuelblum — APPROVED** [review]

**Review by thomasdax98 — APPROVED** [review]

**Pull request merged**